src/modules/UI/scenes/EdgeLogin/EdgeLoginSceneComponent.js

```diff
@@ -21,12 +22,17 @@ type EdgeLoginSceneProps = {
 export default class EdgeLoginScene extends Component<EdgeLoginSceneProps> {
   renderBody (style: Object) {
     let message = this.props.error
+    let textStyle = style.bodyText
     if (!this.props.error) {
       message = s.strings.edge_description
     }
+    if (this.props.lobby && this.props.lobby.loginRequest && this.props.lobby.loginRequest.appId === '') {
```

Comment:
This is because flow does not like it. it will not just let me say if (this.props.lobby.loginRequest.appId)...

Reply:
I understand. But flow is there for a purpose. If the existence of lobby or lobby.loginRequest is optional, then flow will complain. But why is it optional in the first place? If it is optional, then there IS a chance that it doesn't exist and if so, we need to properly handle it.

Reply:
to try to explain it, here is the reducer
```import type { EdgeLobby } from 'edge-core-js'

import * as Constants from '../../constants/indexConstants'
import type { Action } from '../../modules/ReduxTypes.js'

const initialState = {
  lobby: null,
  error: null,
  isProcessing: false
}
export type InitialEdgeLoginState = {
  lobby: null | EdgeLobby,
  error: null | Error,
  isProcessing: boolean
}
export default function (state: InitialEdgeLoginState = initialState, action: Action) {
  switch (action.type) {
    case Constants.PROCESS_ABC_LOGIN:
      return { ...state, isProcessing: true }
    case Constants.INVALIDATE_ABC_LOBBY:
      return { ...state, lobby: null, isProcessing: false }
    case Constants.SET_LOBBY_ERROR:
      return { ...state, lobby: null, error: action.data }
    case Constants.SAVE_ABC_LOBBY:
      return { ...state, lobby: action.data, error: null }
    default:
      return state
  }
}
```


So if there is an error, we are handling it in in the function before the conditional you are talking about

```renderBody (style: Object) {
    let message = this.props.error
    let textStyle = style.bodyText
    if (!this.props.error) {
      message = s.strings.edge_description
    }
    if (this.props.lobby && this.props.lobby.loginRequest && this.props.lobby.loginRequest.appId === '') {
      textStyle = style.bodyTextWarning
      message = sprintf(s.strings.edge_description_warning, this.props.lobby.loginRequest.displayName)
    }
    return (
      <View style={style.body}>
        <Text style={textStyle}>{message}</Text>
      </View>
    )
  }
```

So really the only way based on the reducer we would have a null for the error and for the lobby is if the lobby is invalidated, and that happens only after someone has approved the lobby, and we remove the screen and they login

```export const lobbyLogin = () => async (dispatch: any, getState: any) => {
  const state = getState()
  dispatch(actions.dispatchAction(Constants.PROCESS_ABC_LOGIN))
  await state.core.edgeLogin.lobby.loginRequest.approve()
  dispatch(actions.dispatchAction(Constants.INVALIDATE_ABC_LOBBY))
  Actions.pop()
}
```
The issue we are looking at here is a result of the possibility of their being a null in the reducer, so that is why the property is optional on the component. so I am not sure where you would like me to take this. I can't make the property not optional, but this is one of those screens we can't get to without a error or a lobby in the reducer, so not sure what to do next. Let me know if you want to do a call to discuss further.